lib/wibox/widget/slider.lua

```diff
@@ -458,17 +493,11 @@ function slider:draw(_, cr, width, height)
     x_offset, y_offset = 0, 0
 
     if margins then
-        if type(margins) == "number" then
-            x_offset, y_offset = margins, margins
-            handle_width  = handle_width  - 2*margins
-            handle_height = handle_height - 2*margins
-        else
-            x_offset, y_offset = margins.left or 0, margins.top or 0
-            handle_width  = handle_width  -
-                (margins.left or 0) - (margins.right  or 0)
-            handle_height = handle_height -
-                (margins.top  or 0) - (margins.bottom or 0)
-        end
+        x_offset, y_offset = margins.left or 0, margins.top or 0
+        handle_width  = handle_width  -
+            (margins.left or 0) - (margins.right  or 0)
+        handle_height = handle_height -
+            (margins.top  or 0) - (margins.bottom or 0)
```

Comment:
If we are going to use X times the `margins.direction or 0` statement, we should definitely have local variables to handle these operations once for all.

Reply:
I'm not sure if I fully understand. What scope should the vars be in?

Reply:
Ok, just pushed some changes to the margin stuff. If you could take a look and see if it's what you were looking for.